internal/backend/remote-state/gcs/client.go

```diff
@@ -147,6 +148,15 @@ func (c *remoteClient) lockError(err error) *statemgr.LockError {
 // LockInfo struct.
 func (c *remoteClient) lockInfo() (*statemgr.LockInfo, error) {
 	r, err := c.lockFile().NewReader(c.storageContext)
+
+	// When lock file do exist in Lock() but writing fails because file removed by other process,
```

Comment:
> writing fails

Not sure I understand. Did you mean "reading fails"?

Reply:
Hi, 
writing fails there https://github.com/opentofu/opentofu/blob/36eb93f95862c9f1e8a270f6a6e31f8818b0bfa5/internal/backend/remote-state/gcs/client.go#L104 in Lock() (file exists) then it calls lockInfo() and then reading fails in lockInfo() 
race conditions are hard to explain clearly 
What text should I use there ?

Reply:
So it's "writing fails because of conflict", then "reading fails because the file is no longer there", correct?

Reply:
the sequence is something like 
- lock() checks storage and there is file already (so w.Write fails by condition DoesNotExist:true) 
```
w := lockFile.If(storage.Conditions{DoesNotExist: true}).NewWriter(c.storageContext)
err = func() error {
		if _, err := w.Write(infoJson); err != nil {
			return err
		}
		return w.Close()
	}()
```
- then as error handling, lockInfo() tries to read file 
```
r, err := c.lockFile().NewReader(c.storageContext)
```
and fails too because other process already deleted file

Reply:
Gotcha, I think we're generally on the same page.